web-client/app/scripts/controllers/editProfileController.js

```diff
@@ -37,5 +40,96 @@ angular.module('webClientApp')
       $scope.error = 'حدث خطأ ما.'; // General form error.
       $scope.errorMessages = response.errors; // Detailed error message from backend.
     };
+    
+    /**
+     * Loads all user links
+     */
+    $scope.loadLinks = function() {
+      $scope.links = [{ loading: true }, { loading: true },
+            { loading: true }];
+      UserLink.query({'userId': $routeParams.userId}, function (links) {
+        $scope.links = links;
+      });
+    };
+    
+    $scope.loadLinks();
+
+    /**
+     * Creates a link
+     */
+    $scope.createLink = function(link) {
+      $scope.inProgress = 'createLink';
+      link.userId = $routeParams.userId;
+      Link.save(link, createLinkSuccess, createLinkError);
+    };
+
+    var createLinkSuccess = function() {
+      $analytics.eventTrack('Create Link Success', {
+        category: 'Link'
+      });
+      $scope.inProgress = null;
+    };
+
+    var createLinkError = function(response) {
+      $analytics.eventTrack('Create Link Error', {
+        category: 'Link',
+        label: angular.toJson(response.errors)
+      });
+      $scope.error = 'حدث خطأ في إنشاء الرابط.';
+      $scope.inProgress = null;
+    };
+
+    /**
+     * Updates a link
+     */
+    $scope.updateLink = function(link) {
+      $scope.inProgress = 'updateLink';
+      Link.update({ 'linkId': link.id}, link, updateLinkSuccess, updateLinkError);
+    };
+
+    var updateLinkSuccess = function() {
+      $analytics.eventTrack('Update Link Success', {
+        category: 'Link'
+      });
+      $location.path('/profiles/' + $routeParams.userId);
```

Comment:
This will redirect the user to their profile everytime they update a link. I don't think this should be the case.